src/Debugging/Debugger.php

```diff
@@ -0,0 +1,226 @@
+     *
+     * @var bool
+     */
+    protected bool $showRequest = false;
+
+    /**
+     * Denotes if we send the response to the debugging driver
+     *
+     * @var bool
+     */
+    protected bool $showResponse = false;
+
+    public function __construct()
+    {
+        $this->registerDriver(new RayDebugger);
+        $this->registerDriver(new ErrorLogDebugger);
+        $this->registerDriver(new SystemLogDebugger);
+    }
+
+    /**
+     * Register a driver globally
+     *
+     * @param \Saloon\Contracts\DebuggingDriver $driver
+     *
+     * @return static
+     */
+    public static function registerGlobalDriver(DebuggingDriver $driver): static
+    {
+        static::$globalRegisteredDrivers[$driver->name()] = $driver;
+
+        return new static;
+    }
+
+    /**
+     * Register a driver
+     *
+     * @param \Saloon\Contracts\DebuggingDriver $driver
+     *
+     * @return $this
+     */
+    public function registerDriver(DebuggingDriver $driver): static
+    {
+        $this->registeredDrivers[$driver->name()] = $driver;
+
+        return $this;
+    }
+
+    /**
+     * Subscribe to a given driver
+     *
+     * @param \Saloon\Contracts\DebuggingDriver|string $driver A DebuggingDriver or the name one of a registered one.
+     *
+     * @return $this
+     * @throws \Saloon\Exceptions\UnknownDriverException
+     */
+    public function usingDriver(DebuggingDriver|string $driver): static
+    {
+        if ($driver instanceof DebuggingDriver) {
+            $this->registerDriver($driver);
+        }
+
+        $driverName = is_string($driver) ? $driver : $driver->name();
+
+        // We'll validate that the driver exists
+
+        $registeredDriverNames = array_keys($this->getRegisteredDrivers());
+
+        if (! in_array($driverName, $registeredDriverNames, true)) {
+            throw new UnknownDriverException(sprintf('Unable to find the "%s" driver. Registered drivers: %s', $driverName, implode(', ', $registeredDriverNames)));
+        }
+
+        $this->useDrivers[$driverName] = true;
+
+        return $this;
+    }
+
+    /**
+     * Unsubscribe from a driver
+     *
+     * @param \Saloon\Contracts\DebuggingDriver|string $driver A DebuggingDriver or the name one of a registered one.
+     *
+     * @return $this
+     */
+    public function withoutDriver(DebuggingDriver|string $driver): static
+    {
+        $driverName = is_string($driver) ? $driver : $driver->name();
+
+        unset($this->useDrivers[$driverName]);
+
+        return $this;
+    }
+
+    /**
+     * Only use a given driver
+     *
+     * @param \Saloon\Contracts\DebuggingDriver|string $driver A DebuggingDriver or the name one of a registered one.
+     *
+     * @return $this
+     * @throws \Saloon\Exceptions\UnknownDriverException
+     */
+    public function onlyDriver(DebuggingDriver|string $driver): static
+    {
+        $this->useDrivers = [];
+
+        $this->usingDriver($driver);
+
+        return $this;
+    }
```

Comment:
Also add the driver config callback here, if we add it to the `usingDriver()` method.

```suggestion
    /**
     * Only use a given driver
     *
     * @param \Saloon\Contracts\DebuggingDriver|string $driver A DebuggingDriver or the name one of a registered one.
     * @param callable(\Saloon\Contracts\DebuggingDriver): (void)|null $configureDriver
     *
     * @return $this
     * @throws \Saloon\Exceptions\UnknownDriverException
     */
    public function onlyDriver(DebuggingDriver|string $driver, ?callable $configureDriver = null): static
    {
        $this->useDrivers = [];

        $this->usingDriver($driver, $configureDriver);

        return $this;
    }
```